ods_ci/tests/Tests/500__jupyterhub/custom-image.robot

```diff
@@ -55,26 +57,25 @@ Verify Custom Image Can Be Added
     #Should Match  ${spawner_packages}  ${IMG_PACKAGES}
 
     Spawn Notebook With Arguments  image=${IMAGESTREAM_NAME}  size=Small
-    [Teardown]  Custom Image Teardown
+    ${CLEANUP}=  Set Variable  True
```

Comment:
The reason of using this variable is because you need all the steps in the keyword to be executed before cleaning up? 

If that's the reason, you can use the tags `robot:continue-on-failure` or `robot:recursive-continue-on-failure` to force all steps being executed. There is more info here: https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#continue-on-failure

Reply:
The reason I added it here is to distinguish the behavior of the `Custom Image Teardown`. By default it performs also cleanup operations that are tied with the Jupyter Notebook environment. But this part isn't necessary if the test execution fails before this line. So the default value is set to `false`. Maybe your approach could be used for this too, but I went the way without modifying the current teardown keyword.